tools/LSST_spider_2d_geometry.py

```diff
@@ -0,0 +1,141 @@
+#!/bin/env python3
+
+"""A small tool to convert yaml files from batoid/data to 2D geometries
+which can be used in the imsim.diffraction module."""
+
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional
+from collections.abc import Sequence
+
+from numpy.typing import ArrayLike
+import numpy as np
+
+import batoid
+from batoid import Optic, Baffle, ObscUnion, ObscRectangle, ObscAnnulus
+
+from imsim.diffraction import Geometry
+
+
+def load_spider_geometry(
+    path: Path, strut_thickness: float = 0.1, max_radius=4.4
+) -> Geometry:
+    """Load geometry from a batoid yaml
+
+    Currently, coordinate systems are ignored.
+    """
+    optical_system = Optic.fromYaml(path)
+    baffles = [
+        item.obscuration
+        for item in optical_system.items
+        if isinstance(item, Baffle) and isinstance(item.obscuration, ObscUnion)
+    ]
+    components = set.union(*(set(obscuration.items) for obscuration in baffles))
+    rectangles = [
+        component for component in components if isinstance(component, ObscRectangle)
+    ]
+    rectangles = [
+        rect
+        for rect in rectangles
+        if rect.width < strut_thickness or rect.width < strut_thickness
+    ]
+
+    annuli = [
```

Comment:
I think this is probably missing the `ClearAnnulus` obscuration type (which, to be fair, is not well-documented in batoid).